orttraining/orttraining/python/ort_trainer.py

```diff
@@ -845,6 +883,91 @@ def _prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, i
+        cpu_data = tuple(cpu_data_list)
+
+        # torch inference step
+        with torch.no_grad():
+            torch_out = [self.torch_model_(*cpu_data)]
+        
+        # iterate through outputs, except loss when loss_fn is distinct from the model
+        if self.loss_fn_:
+            torch_out.insert(0, torch.empty(1, 1))
+        for ((out_name, npp_out), pp_out, t_out) in zip(no_postprocess_session_results.items(), postprocess_session_results.values(), torch_out):
+            # the first item of outputs must be 'loss' or some variant
+            if out_name == output_desc[0].name_ and self.loss_fn_:
+                continue
+            no_postprocess_out = npp_out.cpu().numpy().flatten()
+            postprocess_out = pp_out.cpu().numpy().flatten()
+            t_out = t_out.cpu().numpy().flatten()
+            print("Comparing Values of {}".format(out_name))
+            if self._no_postprocess_onnx_model:
+                print("PyTorch vs ONNX before post processing: ", end='')
+                print(np.absolute(np.absolute(no_postprocess_out)-np.absolute(t_out)).mean())
+                print("PyTorch vs ONNX after  post processing: ", end='')
+                print(np.absolute(np.absolute(postprocess_out)-np.absolute(t_out)).mean())
+
+                # compare
+                import math
+                successful_conversion = True
+                for _torch, _ort in zip(t_out.tolist(), no_postprocess_out.tolist()):
+                    if not math.isclose(_torch, _ort, abs_tol=self._tolerance):
+                       warnings.warn("PyTorch model outputs differ from the ORT Model before Post Processing!")
+                       successful_conversion = False
+                       break
+                for _torch, _ort in zip(t_out.tolist(), postprocess_out.tolist()):
+                    if not math.isclose(_torch, _ort, abs_tol=self._tolerance):
+                       warnings.warn("PyTorch model outputs differ from the ORT Model after Post Processing!")
+                       successful_conversion = False
+                       break
+                if successful_conversion:
+                    print("ORT model conversion successful")
+            else:
+                print("PyTorch vs ONNX without post processing: ", end='')
+                print(np.absolute(np.absolute(postprocess_out)-np.absolute(t_out)).mean())
+                
+                # compare
+                import math
+                successful_conversion = True
+                for _torch, _ort in zip(t_out.tolist(), postprocess_out.tolist()):
+                    if not math.isclose(_torch, _ort, abs_tol=self._tolerance):
+                       warnings.warn("PyTorch model outputs differ from the ORT Model without Post Processing!")
+                       successful_conversion = False
+                       break
+                if successful_conversion:
+                    print("ORT model conversion successful")
```

Comment:
This makes the direct comparison without any calculations/normalizations. If this is reasonable, I can reduce duplicate code by making another method